### PR TITLE
move api function out of buttonClick

### PR DESCRIPTION
**Pull request by gswliang**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -48,6 +48,8 @@ export default function App() {
       navigator.geolocation.getCurrentPosition((res) => {
         setLong(res.coords.longitude);
         setLat(res.coords.latitude);
+        getTemp(`${res.coords.latitude},${res.coords.longitude}`);
```
Comment (supaphly2008) [inline src/App.js]: use useRef for long/lat

File: src/App.js [hunk 2]
```diff
@@ -58,8 +60,8 @@ export default function App() {
     init();
   }, []);
 
-  const getTemp = async () => {
-    const { data } = await getTemperature(inputText);
+  const getTemp = async (city) => {
```
Comment (supaphly2008) [inline src/App.js:63]: looks like the parameter is accepting coordinates not city

File: src/WeatherCard.js [hunk 3]
```diff
@@ -49,6 +49,7 @@ const WeatherCard = ({ weather }) => {
   let region = "Taipei";
   let tempCity = "Linkou";
```
Comment (supaphly2008) [inline src/WeatherCard.js:51]: It looks like you are trying to set default values. Replace these with a loading component

File: src/WeatherCard.js [hunk 4]
```diff
@@ -49,6 +49,7 @@ const WeatherCard = ({ weather }) => {
...
 
+  console.log(weather);
   if (weather !== undefined) {
     icon = weather.current.weather_icons[0];
```
Comment (supaphly2008) [inline src/WeatherCard.js]: This won't trigger re-render

File: src/App.js (hunk 1 above)
Comment (gswliang) [reply src/App.js]: resolved

File: src/App.js (hunk 2 above)
Comment (gswliang) [reply src/App.js:63]: resolved

**Pull request merged**